src/genqtclist.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * Tlf - contest logging program for amateur radio operators
+ * Copyright (C) 2013-2014      Ervin Hegedüs - HA2OS <airween@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+	/* ------------------------------------------------------------
+	 *        Generate QTC list to send
+	 *
+	 *--------------------------------------------------------------*/
+
+#include "globalvars.h"
+#include "genqtclist.h"
+
+int genqtclist(char * callsign, int nrofqtc)
+{
+    int qtclistlen;
+    int s = 0, i = 0;
+
+    qtclistlen = 10;
+    if (nrofqtc > 0 && nrofqtc < 10) {
+	qtclistlen = nrofqtc;
+    }
+    qtclist.serial = nr_qtcsent+1;
+    qtclist.marked = 0;
+    qtclist.totalsent = 0;
+    qtclist.count = 0;
+    strncpy(qtclist.callsign, callsign, strlen(callsign));
+    for(s = 0; s < qtclistlen; s++) {
+	qtclist.qtclines[s].qtc[0] = '\0';
+	qtclist.qtclines[s].flag = 0;
+	qtclist.qtclines[s].saved = 0;
+	qtclist.qtclines[s].sent = 0;
+	qtclist.qtclines[s].senttime[0] = '\0';
+    }
+
+    s=next_qtc_qso;
+
+    while (qtclist.count < qtclistlen && s < nr_qsos) {
+        if (strlen(callsign) == 0 || strncmp(qsos[s]+29, callsign, strlen(callsign)) != 0) {	// exclude current callsign
```

Comment:
Why do you ran that loop at all if there is no call sign to look for?
You can test that before the loop.